### PR TITLE
新增国际化功,修复了部分bug

### PR DESCRIPTION
**Pull request by fernandozw**
## For modifications, it is recommended to view it in conjunction with the issue I submitted: https://github.com/SeleniumHQ/selenium-ide/issues/1812
1. There are many files that have been changed for the internationalization function. All the constants involved in label and innerHtml have been replaced with variables.
2. If there is a problem with start electron, it is recommended to delete the /Users/<your name>/Library/Application Support/Electron/config.json file
3. My version is on March 10, 2024. I directly covered the fork version. There will be some differences in page display.
4. Since this project is relatively large and complex, the changes I make will not affect the original functions. Generally, I will add some new methods or variables and will not modify the original logic too much unless I find a bug in the original code.
5. My development environment is Mac OS M1. It is recommended that you execute the pnpm store prune --force and pnpm run clean commands before building.

## 对于修改的地方,建议结合我提交的issue:https://github.com/SeleniumHQ/selenium-ide/issues/1812查看
1、国际化功能改动的文件比较多,把label和innerHtml涉及的常量全部替换成变量了
2、如果start electron 有问题,建议删除/Users/<your name>/Library/Application Support/Electron/config.json文件
3、我的版本是2024年3月10日的,我是直接覆盖了fork的版本,页面展示上会有一些差异
4、由于这个项目比较庞大并且复杂,我涉及改动的地方不会影响原功能,一般我会新增一些方法或者变量,不会去太多的修改原逻辑,除非发现原来的代码有bug
5、我开发的环境是Mac OS M1,建议你在build前,执行pnpm store prune --force 和 pnpm run clean命令

___

## **Type**
enhancement, bug_fix

___

## **Description**
- Introduced internationalization support for the Selenium IDE UI, allowing dynamic language switching between English and Chinese.
- Implemented language maps for UI elements to facilitate easy addition of more languages in the future.
- Refactored the playback controller to improve stability and error handling during test execution.
- Fixed issues related to playback window management, ensuring consistent behavior across different operating systems.
- Added a new method to open and maximize the browser window to a specified URL, enhancing test setup procedures.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>I18N.ts</strong><dd><code>Add Internationalization Support and Language Maps</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
packages/selenium-ide/src/browser/enums/I18N.ts

<li>Added internationalization support for UI elements, including window <br>tabs, Electron menu, file menu, edit menu, view menu, help menu, main <br>page menu, configuration tabs, output configuration page, system <br>configuration page, project configuration page, suites tab, tests tab, <br>splash page, playback page, and test core page.<br> <li> Introduced language maps for Chinese and English, supporting dynamic <br>language switching.<br>

</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium-ide/pull/1813/files#diff-6e6ccd4ae9150222279d1e4dc41464fe4a42b3dbd2444b7aefe4452d4ab828bd">+2176/-0</a></td>

</tr>                    
</table></td></tr><tr><td><strong>Bug_fix
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>index.ts</strong><dd><code>Refactor Playback Controller and Fix Window Handling</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
packages/selenium-ide/src/main/session/controllers/Playback/index.ts

<li>Refactored playback controller to support internationalization.<br> <li> Fixed bugs related to playback window handling and session management.<br> <li> Improved error handling and logging for playback actions.<br> <li> Added a method to open the browser with a specified URL and maximize <br>the window.<br>

</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium-ide/pull/1813/files#diff-cf11b0e5835340a6001040c9a4c6df41c27f6ba86fa06550950df0af284a3215">+239/-201</a></td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: packages/side-runtime/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@seleniumhq/side-runtime",
-  "version": "4.0.5",
+  "version": "4.0.1",
```
Comment (toddtarsi) [inline packages/side-runtime/package.json:3]: @fernandozw - This will become 4.0.6

Comment (toddtarsi) [reply packages/side-runtime/package.json:3]: @fernandozw - Actually, don't worry about that stuff! This is just clutter differences. Nice work! I'll change this stuff later haha.

Comment (fernandozw) [reply packages/side-runtime/package.json:3]: Okay, if you think the function is suitable, you can merge it into the master branch, and then give me developer rights, and we can develop this application together in the future.

Comment (fernandozw) [reply packages/side-runtime/package.json:3]: Actually, I'm a little sorry. When you responded to my issue yesterday, I had already gone to bed. There should be a time difference between us.

Comment (toddtarsi) [reply packages/side-runtime/package.json:3]: Time difference is no issue at all! I really appreciate your help 😄 

I really like your implementation and how you're using getLanguageMap. I didn't know thats how you imported different languages. Very clean use of the api! 👍 

Would you be alright with me cherry picking from your commit? Some of my old stuff is in here and I think all code attribution on it in the history will still have your name ofc 🥇

Comment (fernandozw) [reply packages/side-runtime/package.json:3]: No problem, you can modify it as you like, haha

Comment (fernandozw) [reply packages/side-runtime/package.json:3]: I have discovered several bugs again, and I am not sure if it is caused by my outdated version. After you complete the internationalization process, I will pull the latest branch again and take a look. If it is a bug, I will fix it together and then submit it.

Comment (toddtarsi) [reply packages/side-runtime/package.json:3]: Beta 4 is currently stable, I also have some bugs in latest. Thank you for taking a close eye. This weekend, I will merge in your intl featureset and some fixes, and I will ping you once those are merged in.

Comment (fernandozw) [reply packages/side-runtime/package.json:3]: Okay, I will continue to pay attention to the development of selenium ide, and I will add new features as they become available. Thank you for your reply.

**Comment by toddtarsi** [comment]
@fernandozw - Alright, everything is merged up. For your next PR, it should be a ton more in sync with trunk (as localization was a herculean effort). Great work, and I look forward to seeing what you do next!

**Comment by tuziel** [comment]
I don't understand how i18n works
```typescript
  async getLanguageMap(frontend = false) {
    try {
      const language = await this.session.app.getLocale()

      // what's this? I can't find it in the packages
      const langFile = await import(`../../i18n/${language}/Commands`)
      
      this.languageMap = langFile.default
    } catch (e) {
      // lang DNE, stay en
    } finally {
      if (frontend) {
        // React intl uses a flat dict so we convert it here
        const frontendMap = flattenNestedObject(this.languageMap)
        return frontendMap
      }
      return this.languageMap
    }
  }
```
when I run the app, the variable `language` get `'zh-CN'`, but it still english.

---
我不明白他是怎么工作的
`await import(`../../i18n/${language}/Commands`)`
我找不到这个文件
我跑项目的时候 `language` 是 `zh-CN'`, 但还是显示中文

**Comment by fernandozw** [comment]
toddtarsi 修改了国际化的实现方式,你拉取的应该是最新的代码,我最近1个月太忙没有关注这块的改动,4.0.1-beta.9这个版本是可以的

image

**Comment by fernandozw** [comment]
@tuziel https://github.com/SeleniumHQ/selenium-ide/tree/v4.0.1-beta.9   这个分支,目前来说,trunk分支应该也只是过渡版本,功能还在不断的迭代中,但是核心的录制,回放功能是可以正常使用的,你如果发现任何bug,直接修复,提交pr即可

**Comment by tuziel** [comment]
了解，谢谢大佬抽空回复

**Comment by toddtarsi** [comment]
@fernandozw - Thank you!!!

**Pull request closed without merging**